packages/container/src/render.ts

```diff
@@ -140,11 +141,21 @@ export const composeRenderMethods: ComposeRenderMethodsCallback = ({
           },
           fragmentChildren[0].__k
         );
-      }
+      } else {
+        // Handling for nested or non-root fragments. This will flatten non-root fragments
+
+        return renderedChildren.map((child) => {
+
+          // Return the text content directly if it's a text node
+          if (child.__e instanceof Text) {
```

Comment:
Checking for `Text` doesn't actually work here because `(Fragment instanceof Text) === true`. This Component fails to render because `parseRenderedTree` returns a `props` object instead of a DOM node:
```jsx
<>
  <>
    A<>B</>
  </>
</>
```

Since we need the Fragment to be parsed recursively, this check should only return the string leaf node. I think we do need the `props` string check after all 🙂

Reply:
Very cool, I'll take a look at those links.

> Out of curiosity, how'd you come up with that edge case? This code feels like a good candidate for some fuzz testing 👀

I just needed enough superfluous structure to really exercise the tree logic. It's actually trimmed down from a more pathological test Component 😂 
```jsx
    <>
      <>1</>
      <>
        <br />
        <>
          <input type="text" />
          <>
            A<>B</>
          </>
        </>
      </>
    </>
```

We definitely need some testing around this though, both on the rendered Components and individual modules.

Thanks again @ricky-sb for your help on this!